### PR TITLE
Add `StackedBookGraph`

### PR DESCRIPTION
**Pull request by LRacine**
This is a continuation of the book digraph. Unlike the book digraph, the stacked book digraph has n as the size of the book spine rather than 2. I also changed the book digraph implementation to use this function. For more information on this graph has a look at https://mathworld.wolfram.com/StackedBookGraph.html.

### CODE REVIEW

File: gap/examples.gi [hunk 1]
```diff
@@ -738,3 +736,33 @@ function(filt, m)
...
   return D;
 end);
+
+InstallMethod(StackedBookDigraphCons,
+"for IsMutableDigraph and two positive integers",
+[IsMutableDigraph, IsPosInt, IsPosInt],
+function(filt, m, n)
+  local book;
+  book := ChainDigraph(IsMutable, n);
+  book := DigraphSymmetricClosure(book);
```
Comment (marinaanagno) [inline gap/examples.gi:746]:
```suggestion
  book := DigraphSymmetricClosure(ChainDigraph(IsMutable, n));
```

I thought that maybe a single line would be neater. It works on my computer like that :)

**Review by marinaanagno — COMMENTED** [review]
Cute graph! :) I was googling to see how the graph looks like to get an idea and observed that most sources refer to this graph as the Book **Graph** rather than the Book **Digraph**, which makes sense since it is undirected (or symmetric in digraphs), so maybe it would be better to change the name to `BookGraph` and `StackedBookGraph`? :)

File: gap/examples.gi (hunk 1 above)
Comment (LRacine) [reply gap/examples.gi:746]: Great idea. I will be changing that now! Thank you for the review :D

**Comment by wilfwilson** [comment]
I recommend calling this `StackedBookGraph`, as implicitly discussed on #444 🙂

**Review by mariatsalakou — COMMENTED** [review]
Everything looks good to me

File: doc/examples.xml [hunk 2]
```diff
@@ -711,12 +711,14 @@ true
   <Oper Name="BookDigraph" Arg="[filt, ]m"/>
   <Returns>A digraph.</Returns>
   <Description>
-    The <E>Book digraph</E> is the Cartesian product of a complete digraph with 2 vertices
-    (as the "Book Spine") and the <M><A>m</A>+1</M> star digraph (as the "Pages").
-    For more details on the Book digraph please refer to <URL>https://mathworld.wolfram.com/BookGraph.html</URL>.<P/>
+    The <E>Book digraph</E> is the Cartesian product of the symmetric closure of the
```
Comment (james-d-mitchell) [inline doc/examples.xml:714]: Unless `Book` is someone's name, this should be `book`

File: doc/examples.xml [hunk 3]
```diff
@@ -711,12 +711,14 @@ true
...
+    chain digraph with 2 vertices (as the "Book Spine") and the <M><A>m</A>+1</M> star
```
Comment (james-d-mitchell) [inline doc/examples.xml:715]: "Book spine" should be "book spine"

File: doc/examples.xml [hunk 4]
```diff
@@ -711,12 +711,14 @@ true
...
+    digraph (as the "Pages"). For more details on the Book digraph please refer to
```
Comment (james-d-mitchell) [inline doc/examples.xml:716]: "Pages" -> "pages" + Book -> book

File: doc/examples.xml [hunk 5]
```diff
@@ -743,3 +745,45 @@ true
   </Description>
 </ManSection>
 <#/GAPDoc>
+
+<#GAPDoc Label="StackedBookDigraph">
+<ManSection>
+  <Oper Name="StackedBookDigraph" Arg="[filt, ]m, n"/>
+  <Returns>A digraph.</Returns>
+  <Description>
+    The <E>Stacked Book digraph</E> is the Cartesian product of the symmetric closure of the
```
Comment (james-d-mitchell) [inline doc/examples.xml:754]: Capitalization

File: doc/examples.xml [hunk 6]
```diff
@@ -743,3 +745,45 @@ true
...
+    chain digraph with <A>n</A> vertices (as the "Book Spine") and the <M><A>m</A>+1</M> star digraph
```
Comment (james-d-mitchell) [inline doc/examples.xml:755]: Ditto

**Review by james-d-mitchell — CHANGES_REQUESTED** [review]
This looks good apart from some inappropriate capitalization in the doc, and possibly modulo the names of the functions. I'll comment on that elsewhere (if appropriate)!.

**Comment by wilfwilson** [comment]
This now has merged conflicts because of #486  being merged.

File: doc/examples.xml [hunk 7]
```diff
@@ -709,14 +709,16 @@ true
 <#GAPDoc Label="BookGraph">
 <ManSection>
   <Oper Name="BookGraph" Arg="[filt, ]m"/>
-  <Returns>A digraph.</Returns>
+  <Returns>A graph.</Returns>
```
Comment (wilfwilson) [inline doc/examples.xml:712]: The `Returns` tag should state the type of GAP object that is returned. So here, even though it's mathematically a graph, it is in practice a digraph object that you get back, so this should indeed still say:
```suggestion
  <Returns>A digraph.</Returns>
```

File: doc/examples.xml [hunk 8]
```diff
@@ -743,3 +745,46 @@ true
   </Description>
 </ManSection>
 <#/GAPDoc>
+
+<#GAPDoc Label="StackedBookGraph">
+<ManSection>
+  <Oper Name="StackedBookGraph" Arg="[filt, ]m, n"/>
+  <Returns>A graph.</Returns>
```
Comment (wilfwilson) [inline doc/examples.xml:752]: As should this one:
```suggestion
  <Returns>A digraph.</Returns>
```

File: doc/examples.xml [hunk 9]
```diff
@@ -743,3 +745,46 @@ true
...
+  <Description>
+    The <E>Stacked book digraph</E> is the Cartesian product of the symmetric closure of the
```
Comment (wilfwilson) [inline doc/examples.xml:754]: Looks like there's another 😉 
```suggestion
    The <E>stacked book digraph</E> is the Cartesian product of the symmetric closure of the
```

File: doc/examples.xml [hunk 10]
```diff
@@ -743,3 +745,46 @@ true
...
+    chain digraph with <A>n</A> vertices (as the "book spine") and the <M><A>m</A>+1</M> star digraph
+    (as the "pages"). For more details on the Stacked book digraph please refer to
```
Comment (wilfwilson) [inline doc/examples.xml:756]: or two!
```suggestion
    (as the "pages"). For more details on the stacked book digraph please refer to
```

**Review by wilfwilson — APPROVED** [review]
Thanks for all these changes!

**Comment by wilfwilson** [comment]
I've squashed everything into one commit and resolved the merge conflicts, so I'll merge once the tests pass for a final time 🙂 . Thanks!

**Pull request merged**